From 5f478733715e1a09b5040fe423e9238085875e8a Mon Sep 17 00:00:00 2001
From: Eric Ruei <e-ruei1@ti.com>
Date: Fri, 30 Aug 2019 18:06:15 -0400
Subject: [PATCH] HACK: gen_dispatch: use <KHR/khrplatform.h> instead of local
 definitions

The common <KHR/khrplatform.h> should be used instead of local definitions

Signed-off-by: Eric Ruei <e-ruei1@ti.com>
---
 src/gen_dispatch.py | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/src/gen_dispatch.py b/src/gen_dispatch.py
index 28dad2b..5458414 100755
--- a/src/gen_dispatch.py
+++ b/src/gen_dispatch.py
@@ -506,28 +506,17 @@ class Generator(object):
             # don't directly include khrplatform.h because it's not
             # present on many systems, and coming up with #ifdefs to
             # decide when it's not present would be hard.
-            self.outln('#define __khrplatform_h_ 1')
             self.outln('typedef int8_t khronos_int8_t;')
             self.outln('typedef int16_t khronos_int16_t;')
-            self.outln('typedef int32_t khronos_int32_t;')
-            self.outln('typedef int64_t khronos_int64_t;')
             self.outln('typedef uint8_t khronos_uint8_t;')
             self.outln('typedef uint16_t khronos_uint16_t;')
-            self.outln('typedef uint32_t khronos_uint32_t;')
-            self.outln('typedef uint64_t khronos_uint64_t;')
             self.outln('typedef float khronos_float_t;')
             self.outln('typedef long khronos_intptr_t;')
             self.outln('typedef long khronos_ssize_t;')
             self.outln('typedef unsigned long khronos_usize_t;')
             self.outln('typedef uint64_t khronos_utime_nanoseconds_t;')
             self.outln('typedef int64_t khronos_stime_nanoseconds_t;')
-            self.outln('#define KHRONOS_MAX_ENUM 0x7FFFFFFF')
-            self.outln('typedef enum {')
-            self.outln('    KHRONOS_FALSE = 0,')
-            self.outln('    KHRONOS_TRUE  = 1,')
-            self.outln('    KHRONOS_BOOLEAN_ENUM_FORCE_SIZE = KHRONOS_MAX_ENUM')
-            self.outln('} khronos_boolean_enum_t;')
-            self.outln('typedef uintptr_t khronos_uintptr_t;')
+            self.outln('#include <KHR/khrplatform.h>')
 
         if self.target == "glx":
             self.outln('#include <X11/Xlib.h>')
-- 
2.17.1

